vpr/src/route/connection_box_lookahead_map.cpp

```diff
@@ -213,6 +228,404 @@ class CostMap {
 
 static CostMap g_cost_map;
 
+// Search grid used for sampling segments from graph.
+//
+// SearchGrid divides the graph based on segment type, and within each
+// segment type defines a NxN uniform search grid, where N is kTargetGridSize.
+//
+// SearchGrid needs to be initialized with the setup and add_node methods.
+//
+// Once initialized, next_node can be invoked to determine next rr node to
+// expand.
+//
+// During expansion, visit_entry and expanded should be called to indicate
+// that an expansion found data relevant to the search.
+class SearchGrid {
+    // Representation of one search area of the grid.
+    struct GridEntry {
+        GridEntry()
+            : samples(0)
+            , bounds(vtr::Rect<int>(-1, -1, -1, -1)) {}
```

Comment:
Both `vtr::Rect` and `t_bb` are used to represent bounding boxes.

Reply:
It's all about whether the upper bound should be inclusive or exclusive.